src/linker/Linker/TypeMapInfo.cs

```diff
@@ -299,9 +301,20 @@ IEnumerable<InterfaceImplementation> GetDefaultInterfaceImplementations (TypeDef
 
 		MethodDefinition TryMatchMethod (TypeReference type, MethodReference method)
 		{
+			var isStaticInterfaceMethod = context.TryResolve (method)?.IsStatic;
```

Comment:
I wouldn't expect us to run TryMatchMethod for static interface methods. Those are not implemented by a name/sig match and this search would return bogus results:

```csharp
interface IFoo
{
    static virtual void Blah();
}

class Foo : IFoo
{
    // This method is unrelated to IFoo.Blah
    public static void Blah() => throw null;

    // This one implements it
    static void IFoo.Blah() => throw null;
}
```

We should have an early out for static methods before this path is reached.